src/Audio_details.md

```diff
@@ -0,0 +1,192 @@
+  rect.legend { width: 5px; height: 5px; }
+  text.legend { font-size: .7em; }
+  .arrow { stroke-width: 2px; opacity: .35; }
+  .unfilled {
+    fill: none;
+  }
+  .no-stroke {
+    stroke: none;
+  }
+</style>
+
+<figure>
+{{#include imgs/apu_detailed.svg:2:}}
+
+<figcaption>Source: Lior "LIJI32" Halphon</figcaption>
+</figure>
+
+Each of the four "conceptual" channels is composed of a "generation" circuit (designated "channel" in the above diagram), and a [DAC](https://en.wikipedia.org/wiki/Digital-to-analog_converter).
+The digital value produced by the generator, which ranges between $0 and $F (0 and 15), is linearly translated by the DAC into an analog[^digital_analog] value between -1 and 1 (the unit is arbitrary).
+
+The four analog channel outputs are then fed into the mixer[^vin], which selectively adds them (depending on [`NR51`](<#FF25 — NR51: Sound panning>)) into two analog outputs (Left and Right).
+Thus, the analog range of those outputs is 4× that of each channel, -4 to 4.
+
+Then, both of these two get their respective volume scaled, once from [`NR50`](<#FF24 — NR50: Master volume & VIN panning>), and once from the volume knob (if the console has one).
+Note that the former step never mutes a non-silent input, but the latter can.
+
+Each of the two analog outputs then goes through a [high-pass filter](https://en.wikipedia.org/wiki/High-pass_filter) (HPF).
+For short, a HPF constantly tries to "pull" the signal towards analog 0 (neutral); the reason for that is explained further below.
+
+[^digital_analog]:
+To be clear: digital values are discrete and clear-cut; conversely, the analog domain is continuous.
+The former is what computers use, the latter is what the real world is made of.
+
+[^vin]:
+Actually, VIN acts as a 5<sup>th</sup> channel fed into the mixer, whose control bits are in NR50 instead of NR51.
+This was omitted from the diagram for simplicity.
+
+## PCM registers
+
+These two registers, only present in the Game Boy Color and ulterior models, allow reading the output of the generation circuits directly; this is very useful for testing "internal" APU behavior.
+These registers are not documented in any known Nintendo manual.
+
+### FF76 — PCM12 (CGB Mode only): Digital outputs 1 & 2 \[read-only\]
+
+The low nibble is a copy of sound channel 1's digital output, the high nibble a copy of sound channel 2's.
+
+### FF77 — PCM34 (CGB Mode only): Digital outputs 3 & 4 \[read-only\]
+
+Same, but with channels 3 and 4.
+
+## Finer technical explanation
+
+### DIV-APU
+
+A "DIV-APU" counter is increased every time `DIV`'s bit 4 (5 in [double-speed mode](<#FF4D — KEY1 (CGB Mode only): Prepare speed switch>)) goes from 1 to 0, therefore at a frequency of 512 Hz (regardless of whether double-speed is active).
+Thus, the counter can be made to increase faster by writing to `DIV` while its relevant bit is set (which clears `DIV`, and triggers the falling edge).
+
+Some events are tied to the DIV-APU counter:
```

Comment:
```suggestion
The rate of the following events are tied to the DIV-APU counter:
```

Reply:
It's not just the rate, it's their occurrence. Repeatedly writing to DIV allows triggering them irregularly, or not at all.

Reply:
Which is what rate conveys here. If I understand this correctly, it's not really "irregular", it's still every *N* ticks. Then the fact that those ticks can happen or not it's a separate concept. The event A is still going to have a rate of 1/N DIV-APU ticks.

Reply:
```suggestion
The following events occur every <var>N</var> DIV-APU ticks:
```
Then we'd replace the "Every N ticks" column header with "Rate".

Reply:
It sounds better to me